src/components/subs/CardImage.tsx

```diff
@@ -1,7 +1,221 @@
+        <image
+          key={`card-rarity-${i}`}
+          href={rarityIcon}
+          x={i * 72 + 16}
+          y="490"
+          width="72"
+          height="70"
+        />
+      ))}
+    </svg>
+  ) : (
+    skeleton
+  );
 };
 
-export default CardImage;
+export const CardSmallImage: React.FC<{ id: number }> = ({ id }) => {
+  const skeleton = CardImageSkeleton({});
+
+  const classes = useSvgStyles();
+  const [cards] = useCachedData<ICardInfo>("cards");
+  const [card, setCard] = useState<ICardInfo>();
+
+  const [hoveredArea, setHoveredArea] = useState<number>(0);
+  const [imgLeftX, refImgLeftX, setImgLeftX] = useRefState<number>(-256);
+  const [imgRightX, refImgRightX, setImgRightX] = useRefState<number>(512);
+
+  const rarityIcon =
+    card && card.rarity >= 3 ? rarityAfterTraining : rarityNormal;
+
+  useEffect(() => {
+    if (cards.length) setCard(cards.find((elem) => elem.id === id));
+  }, [cards, id]);
+
+  useEffect(() => {
+    if (hoveredArea === 2) {
+      const interval = setInterval(() => {
+        if (refImgLeftX.current < 0) {
+          setImgLeftX((x) => Math.floor(x + 10 * (-x / 256)));
+          setImgRightX((x) => Math.floor(x + 10 * ((768 - x) / 256)));
+        } else {
+          clearInterval(interval);
+        }
+      }, 5);
+      return () => clearInterval(interval);
+    } else if (hoveredArea === 1) {
+      const interval = setInterval(() => {
+        if (refImgRightX.current > 256) {
+          setImgLeftX((x) => Math.floor(x - 10 * ((512 + x) / 256)));
+          setImgRightX((x) => Math.floor(x - 10 * ((x - 256) / 256)));
+        } else {
+          clearInterval(interval);
+        }
+      }, 5);
+      return () => clearInterval(interval);
+    } else if (hoveredArea === 0) {
+      const interval = setInterval(() => {
+        if (refImgLeftX.current > -256) {
+          setImgLeftX((x) => Math.floor(x - 10 * ((256 + x) / 256)));
+          setImgRightX((x) => Math.floor(x - 10 * ((x - 512) / 256)));
+        } else if (refImgLeftX.current < -256) {
+          setImgLeftX((x) => Math.floor(x + 10 * ((-256 - x) / 256)));
+          setImgRightX((x) => Math.floor(x + 10 * ((512 - x) / 256)));
+        } else {
+          clearInterval(interval);
+        }
+      }, 5);
+      return () => clearInterval(interval);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hoveredArea]);
+
+  return card ? (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox="0 0 1024 576"
+      className={classes.svg}
+      onMouseLeave={() => setHoveredArea(0)}
+    >
+      {card.rarity >= 3 ? (
+        <Fragment>
+          <svg
+            x={imgRightX}
+            y="0"
+            width="768"
+            height="576"
+            preserveAspectRatio="xMaxYMid slice"
+            viewBox="0 0 1024 576"
+            onMouseOver={() => setHoveredArea(1)}
+            style={{ pointerEvents: "all" }}
+          >
+            <image
```

Comment:
because `width` attribute is required,
```suggestion
            <image
              width="1024"
```